driver-async/src/main/com/mongodb/async/client/MongoIterableSubscription.java

```diff
@@ -108,12 +106,15 @@ public void onResult(final List<TResult> result, final Throwable t) {
     /**
      * Returns the batchSize to be used with the cursor.
      *
-     * <p>Anything less than 2 would close the cursor so that is the minimum batchSize and `Integer.MAX_VALUE` is the maximum
-     * batchSize.</p>
+     * <p>If the batch size has been set on the MongoIterable that is used, otherwise the requested demand is used. When using requested
```

Comment:
While making sure I fully understood the batch size issue, I had another look at this, and now wonder whether it would be better to use the max of mongoIterable.getBatchSize() and getRequested() (if batchSize is not null, that is).

Reply:
No, that wouldn't deliver the fine grain control that has been asked for in the ticket:

> Composition libraries request by default Long.MAX_SIZE or follow custom request patterns (such as a filter operator that requests request(1) if a particular element does not match the filter criteria). A configurable batchSize would make sense to prevent requests with small batch sizes and to ensure evenly-sized chunks of data.

Reply:
OK, I see.